symphony/lib/toolkit/class.widget.php

```diff
@@ -403,7 +403,7 @@ public static function TableHead(array $columns = null)
                     $th->setAttribute('scope', $col[1]);
                 }
 
-                if (is_array($col[2]) && !empty($col[2])) {
+                if (!empty($col[2]) && is_array($col[2])) {
```

Comment:
`if (General::isNotAnEmptyArray($col[2]))` would hide the `is_array` part. (I would expect this function to return `true` for a `null` value as well, because `null` *is not an empty array*.) So we would have to call it `General::isNotEmptyAndIsAnArray($col[2])`. :-)

Anyway, using a function of the `General` class would only help experienced developers. Others wouldn't use them (and, again, write stuff like `is_array($col[2]) && !empty($col[2])`). (Take me, for example: Working with Symphony taught me everything I know about PHP coding.) If we want people to grow with Symphony, I suggest to find *the best way to deal with certain things* and use these solutions as visible (explicit) as possible. (Is there any "official recommendation" for certain patterns?)